src/pages/GradingBook.tsx

```diff
@@ -0,0 +1,64 @@
+import { api } from "api"
+import { useApiCall } from "api/useApiCall"
+import { GradingBookCareerInfo } from "components/GradingHook/GradingBookCareerInfo"
+import { ProgressCircleCareer } from "components/GradingHook/ProgressCircleCareer"
+import { PageWrap } from "components/PageLayout"
+import { Text } from "components/Text"
+import { LoginContext } from "contexts/login"
+import { MainStackScreen } from "navigation/NavigationTypes"
+import { useContext, useMemo } from "react"
+import { useTranslation } from "react-i18next"
+import { StyleSheet } from "react-native"
+import { palette, usePalette } from "utils/colors"
+
+export const GradingBook: MainStackScreen<"GradingBook"> = () => {
+  const { isLight } = usePalette()
+  const { loggedIn, userInfo } = useContext(LoginContext)
+  const { t } = useTranslation("gradingBook")
+
+  const styles = useMemo(
+    () =>
+      StyleSheet.create({
+        careerSubtitle: {
+          fontSize: 20,
+          textAlign: "center",
+          fontWeight: "500",
+          color: isLight ? palette.primary : "#ffffff",
+        },
+      }),
+    [isLight]
+  )
+
+  const { matricola } = userInfo?.careers?.[0] ?? {}
+
+  const [careerStats] = useApiCall(
+    api.gradingbook.getGradingBook,
+    { matricola: matricola ?? "" },
+    [loggedIn],
+    {},
+    !loggedIn
+  )
+
+  if (!careerStats) return null
+
```

Comment:
invece di restituire null, che fa renderizzare una pagina completamente bianca, renderizza comunque la pagina con il titolo ma mostra un avviso di caricamento / errore

per esempio si potrebbe fare il caricamento tipo:

```suggestion
  const [careerStats, loading] = useApiCall(
    api.gradingbook.getGradingBook,
    { matricola: matricola ?? "" },
    [loggedIn],
    {},
    !loggedIn
  )

  return (
    <PageWrap title={t("gradingBook_title")}>
      {loading ? (
        <LoadingIndicator />
      ) : <>
          <Text>
          ...
        </>
      }
  )
```

e poi se careerStats è null anche mentre non sta caricando puoi mostrare un ErrorMessage col bel faccino di Polifemo